src/main/java/com/flowingcode/vaadin/addons/twincolgrid/TwinColGrid.java

```diff
@@ -455,49 +454,62 @@ private void updateSelection(final Set<T> addedItems, final Set<T> removedItems)
 		});
 	}
 
-	@SuppressWarnings("unchecked")
 	private void configDragAndDrop(final Grid<T> sourceGrid, final Grid<T> targetGrid) {

```

Comment:
Change the method signature to `configDragAndDrop(TwinColModel<T> source, TwinColModel<T> target)` in order to avoid the conditional for determining which one is the sourceModel.